### PR TITLE
Feature: Introduce NavigationViewItemButtonStyle to Settings dialog

### PR DESCRIPTION
**Pull request by onein528**
### Description

Introducing new NavView style to Settings pages. (In another PR, to Properties pages)

### Motivation and Context

Basing on a new concept style of NavigationView, I am going to make them real.

### Details of changes

- Add NavigationViewButtonStyle
- Refactor SettingsDialog
- Add missing namespace 'Views' in 'Files.App.Settings'

### Validation

- [x] Built and ran the app
- [x] Tested the changes for accessibility

#### Main style

image

#### Collapsed style

image

### CODE REVIEW

**Comment by yaira2** [comment]
> Refactor SettingsDialog

Depending what you mean by this, we might want to split it into a different PR.

**Comment by yaira2** [comment]
> Refactor XAML in settings pages

This should be a separate PR and should be discussed beforehand.

File: src/Files.App/Dialogs/SettingsDialog.xaml [hunk 1]
```diff
@@ -79,7 +84,7 @@
 					</NavigationViewItem>
 					<NavigationViewItem
 						AccessKey="P"
-						AutomationProperties.AutomationId="SettingsItemPreferences"
+						AutomationProperties.AutomationId="MainSettingsNavigationViewItemPreferences"
```
Comment (yaira2) [inline src/Files.App/Dialogs/SettingsDialog.xaml:87]: The automated tests need to be updated to match this.

**Comment by onein528** [comment]
Ohh messed up again. I should create new one that is limited the scope.

**Comment by yaira2** [comment]
> Ohh messed up again. I should create new one that is limited the scope.

I'll review it because the work is done but keep in mind for the future 🙂

**Comment by onein528** [comment]
I'm reverting some super unrelated changes

File: src/Files.App/Views/Settings/TagsPage.xaml [hunk 2]
```diff
@@ -24,19 +25,19 @@
...
 		<StackPanel
 			HorizontalAlignment="Stretch"
 			VerticalAlignment="Stretch"
 			Spacing="4">
 			<StackPanel.ChildrenTransitions>
-				<TransitionCollection>
-					<EntranceThemeTransition />
-				</TransitionCollection>
+				<EntranceThemeTransition FromVerticalOffset="50" />
+				<RepositionThemeTransition IsStaggeringEnabled="False" />
```
Comment (yaira2) [inline src/Files.App/Views/Settings/TagsPage.xaml:40]: Let's split this into another PR, it's going to take me some time to test and see if I want to use the change.

File: src/Files.App/Views/Settings/PreferencesPage.xaml.cs [hunk 3]
```diff
@@ -1,12 +1,10 @@
...
 
-namespace Files.App.Settings
+namespace Files.App.Views.Settings
 {
 	public sealed partial class PreferencesPage : Page
 	{
 		public PreferencesPage()
-		{
-			InitializeComponent();
-		}
```
Comment (yaira2) [inline src/Files.App/Views/Settings/PreferencesPage.xaml.cs:10]: Brackets should be kept

File: src/Files.App/Dialogs/SettingsDialog.xaml [hunk 4]
```diff
@@ -20,18 +26,14 @@
...
 				<RowDefinition Height="44" />
 				<RowDefinition Height="*" />
 			</Grid.RowDefinitions>
 
-			<Grid
-				Grid.Row="0"
-				Padding="8,0"
-				HorizontalAlignment="Stretch">
+			<!--  Titlebar  -->
+			<Grid Padding="8,0">
```
Comment (yaira2) [inline src/Files.App/Dialogs/SettingsDialog.xaml:36]: We always define the row, makes it easier to read the xaml.

**Comment by mdtauk** [comment]
This control template should be ready for the new Icon once it rolls out fully.

And Overflow scenarios need to be considered too.  I have put together a design in the Figma for this control pattern

**Comment by onein528** [comment]
I'm not sure I do understand what LeftPaneOverflow style is. @mdtauk What's that?

**Comment by yaira2** [comment]
@onein528 don't worry about this, the scrollbar takes care of this scenario.

**Comment by onein528** [comment]
Once this is merged, I will work on properties window as well.

File: src/Files.App/Dialogs/SettingsDialog.xaml [hunk 5]
```diff
@@ -40,32 +42,34 @@
...
-				x:Name="SettingsPane"
+				x:Name="MainSettingsNavigationView"
 				Grid.Row="1"
 				IsBackButtonVisible="Collapsed"
 				IsBackEnabled="False"
 				IsPaneToggleButtonVisible="False"
 				IsSettingsVisible="False"
 				IsTitleBarAutoPaddingEnabled="False"
-				OpenPaneLength="260"
+				OpenPaneLength="224"
```
Comment (yaira2) [inline src/Files.App/Dialogs/SettingsDialog.xaml:68]: 260 is good, works for different localizations.

File: src/Files.App/Dialogs/SettingsDialog.xaml.cs [hunk 6]
```diff
@@ -10,38 +10,33 @@ namespace Files.App.Dialogs
...
 			App.Window.SizeChanged += Current_SizeChanged;
 			UpdateDialogLayout();
 		}
 
-		public new async Task<DialogResult> ShowAsync() => (DialogResult)await base.ShowAsync();
+		public new async Task<DialogResult> ShowAsync()
+			=> (DialogResult)await base.ShowAsync();
 
 		private void Current_SizeChanged(object sender, WindowSizeChangedEventArgs e)
-		{
```
Comment (yaira2) [inline src/Files.App/Dialogs/SettingsDialog.xaml.cs:33]: Brackets...

File: src/Files.App/Dialogs/SettingsDialog.xaml.cs [hunk 7]
```diff
@@ -63,9 +58,7 @@ private void ContentDialog_Closing(ContentDialog sender, ContentDialogClosingEve
 			App.Window.SizeChanged -= Current_SizeChanged;
 		}
 
-		private void ButtonClose_Click(object sender, RoutedEventArgs e)
-		{
-			Hide();
-		}
+		private void CloseSettingsDialogButton_Click(object sender, RoutedEventArgs e)
+			=> Hide();
```
Comment (yaira2) [inline src/Files.App/Dialogs/SettingsDialog.xaml.cs:62]: Brackets

**Comment by yaira2** [comment]
Happens when there are other pipelines running at the same time.

**Comment by onein528** [comment]
Now, NavViewItem has completely the same style as Button one.

File: src/Files.App/Dialogs/SettingsDialog.xaml [hunk 8]
```diff
@@ -6,133 +6,135 @@
...
-					<Frame x:Name="SettingsContentFrame" Padding="12" />
-				</ScrollViewer>
-			</NavigationView>
-		</Grid>
-	</Border>
+			<!--  Content Frame  -->
+			<ScrollViewer x:Name="SettingsContentScrollViewer">
+				<Frame
+					x:Name="SettingsContentFrame"
+					Padding="16"
```
Comment (yaira2) [inline src/Files.App/Dialogs/SettingsDialog.xaml:134]: Why was this changed?

Comment (onein528) [reply src/Files.App/Dialogs/SettingsDialog.xaml:134]: Actually, should be 24px and 16px for small window

image

Comment (yaira2) [reply src/Files.App/Dialogs/SettingsDialog.xaml:134]: This is a small window if you measure the width of the page.

Comment (onein528) [reply src/Files.App/Dialogs/SettingsDialog.xaml]: @mdtauk Yeah I agree. At this time, we should limit the scope of purpose of the PR.
